classes/app.rb

```diff
@@ -0,0 +1,137 @@
+require_relative 'book'
+require_relative 'classroom'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_books_with_index
+    @books.each_with_index { |book, i| puts "#{i}) Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_people_with_index
+    @people.each_with_index do |person, i|
+      puts "#{i}) [#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    student_or_teacher = gets.chomp.to_i
+    case student_or_teacher
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts "Invalid choice. Please enter a valid option. (#{student_or_teacher})"
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp.to_s
+
+    print 'Has parent permission? [Y / N]: '
+    parent_permission = gets.chomp.to_s
```

Comment:
- [ ] 🎓👨‍🏫👩‍🏫 To follow the single responsibility principle effectively, all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. consider below as an example to help you refactor further

```sh
  private

  def prompt_input(message)
    print message
    gets.chomp
  end

  def prompt_yes_no(message)
    val = prompt_input("#{message} (Y/N): ").downcase
    %w[y yes].include?(val)
  end
```